engine/src/main/java/org/hibernate/search/query/dsl/sort/SortNativeContext.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.query.dsl.sort;
+
+/**
+ * @author Emmanuel Bernard emmanuel@hibernate.org
+ */
+public interface SortNativeContext extends SortAdditionalSortFieldContext, SortTermination {
```

Comment:
Ok that sounds compelling, but currently we can't eliminate `createFullTextQuery` as our query DSL only covers building a handful of queries.

So your suggestion is that in a next step we'd need to provide a full alternative to `createFullTextQuery` able to produce any advanced & custom query... essentially a way to include "native query definitions" in our DSL in a similar and consistent way with the Sort DSL proposal?

That sounds nice and I'm not against it, but I'm surprised that you'd start such a direction from the sort DSL.